### PR TITLE
[vcpkg script] vcpkg_configure_meson: Support wasm32-emscripten triplet

### PR DESCRIPTION
**Pull request by past-due**
**Describe the pull request**

Fixes "Unsupported target architecture" error when trying to compile meson-based ports targeting the `wasm32-emscripten` triplet.

Tested this with the fribidi port. (Compiled, linked successfully with example project, executed.)

### CODE REVIEW

**Comment by past-due** [comment]
@LilyWangLL Is there anything remaining that's needed to get this merged? (On my end this works well, has been tested end-to-end with the `wasm32-emscripten` triplet with the fribidi port, and is working great.)

**Comment by past-due** [comment]
@LilyWangLL From what I can see, the failures do not appear to have anything to do with this PR. (The ports listed don't even use meson.)

**Comment by LilyWangLL** [comment]
Depends on PR #26646

**Review by LilyWangLL — APPROVED** [review]

File: scripts/cmake/vcpkg_configure_meson.cmake [hunk 1]
```diff
@@ -259,6 +259,9 @@ function(z_vcpkg_meson_generate_cross_file additional_binaries) #https://mesonbu
     elseif(VCPKG_TARGET_ARCHITECTURE MATCHES "loongarch64")
         set(host_cpu_fam loongarch64)
         set(host_cpu loongarch64)
+    elseif(VCPKG_TARGET_ARCHITECTURE MATCHES "wasm32")
```
Comment (dan-shaw) [inline scripts/cmake/vcpkg_configure_meson.cmake:262]: Does the `${MACHINE}` variable need to be updated as well?

Comment (past-due) [reply scripts/cmake/vcpkg_configure_meson.cmake:262]: @dan-shaw: Not based on my tests.

I've done end-to-end testing with the `fribidi` port - this PR fixes things so it compiles, links successfully with an example project, and functions properly.

**Pull request merged**